app/models/hubstats/deploy.rb

```diff
@@ -1,6 +1,14 @@
 module Hubstats
   class Deploy < ActiveRecord::Base
 
+    before_validation :check_time, on: :create
+    validates :git_revision, :deployed_at, :deployed_by, :repo_id, presence: true
+    validates_associated :repo
```

Comment:
This isn't providing any value since Repo doesn't have any validations itself.  I would vote for removing it since its currently extraneous overhead.